api/v1beta2/types.go

```diff
@@ -163,3 +193,20 @@ type VPCEndpoint struct {
 	// +optional
 	LBID *string `json:"loadBalancerIPID,omitempty"`
 }
+
+// VPCResource represents a specific VPC resource.
+// +kubebuilder:validation:XValidation:rule="!has(self.id) && !has(self.name)",message="an id or name must be provided"
+type VPCResource struct {
+	// id of the resource.
+	// +kubebuilder:validation:MinLength=1
+	// +optional
+	ID *string `json:"id,omitempty"`
+
+	// name of the resource.
+	// +kubebuilder:validation:MinLength=1
+	// +optional
+	Name *string `json:"name,omitempty"`
+
+	// type is the type of VPC resource.
+	Type *ResourceType `json:"type,omitempty"`
```

Comment:
Would like to understand how this will be used. any example?

Reply:
A VPCResource can be any kind of VPC resource (VSI, LB, SG, SGR, etc.).
Since I would only have a name or ID, it is helpful to have additional details as to what type of resource it is, IMHO.

Reply:
I dropped Type for now.

Reply:
same comments as above, if you have a plan on how to use that ID in controller that might help us in understanding this better.